calculator.js

```diff
@@ -1,5 +1,223 @@
+/// BASE CASE SAMPLE INPUTS ///
+
 const exampleAdditionInput = {
   num1: 3,
   num2: 5,
-  operation: 'add',
+  op: 'Add',
 }
+
+const exampleAdditionInputSign = {
+  num1: 3,
+  num2: 5,
+  op: '+',
+}
+
+const exampleSubtractInput = {
+  num1: 10,
+  num2: 5,
+  op: 'subtract',
+}
+
+const exampleSubtractInputSign = {
+  num1: 10,
+  num2: 5,
+  op: '-',
+}
+
+const exampleMultiInput = {
+  num1: 3,
+  num2: 5,
+  op: 'multiply',
+}
+
+const exampleMultiInputSign = {
+  num1: 3,
+  num2: 5,
+  op: '*',
+}
+
+const exampleDivideInput = {
+  num1: 3,
+  num2: 5,
+  op: 'divide',
+}
+
+const exampleDivideInputSign = {
+  num1: 3,
+  num2: 5,
+  op: '/',
+}
+
+const exampleModuloInput = {
+  num1: 5,
+  num2: 3,
+  op: 'modulo',
+}
+
+const exampleModuloInputSign = {
+  num1: 5,
+  num2: 3,
+  op: '%',
+}
+
+const exampleExponentInput = {
+  num1: 2,
+  num2: 3,
+  op: 'exponent',
+}
+
+const exampleExponentInputSign = {
+  num1: 2,
+  num2: 3,
+  op: '**',
+}
+
+/// EDGE CASE SAMPLE INPUTS ///
+
+const exampleMissingNum = {
+  num1: null,
+  num2: 5,
+  op: '+',
+}
+
+const exampleMissingOp = {
+  num1: 3,
+  num2: 5,
+  op: null,
+}
+
+const exampleWrongSign = {
+  num1: 3,
+  num2: 5,
+  op: '(',
+}
+
+const exampleOneNonNum = {
+  num1: 'hotdog',
+  num2: 5,
+  op: '/',
+}
+
+const exampleTwoNonNum = {
+  num1: 'hotdog',
+  num2: 'elephant',
+  op: '*',
+}
+
+const exampleDivideByZero = {
+  num1: 5,
+  num2: 0,
+  op: 'divide',
+}
+
+/// CALCULATOR FUNCTIONALITY ///
+const operator = {
+  add: '+',
+  subtract: '-',
+  divide: '/',
+  multiply: '*',
+  exponent: '**',
+  modulo: '%',
+}
+
+function verifyNum(input) {
+  return isNaN(input) || !input
+}
+
+function printNum(data, result) {
+  console.log(`${data.num1} ${operator[data.op.toLowerCase()] || data.op} ${data.num2} = ${result}`)
+}
+
+function calculator(data) {
+  let sign = String(data.op)
+  sign = operator[sign.toLowerCase()] || sign
+  
+  if (verifyNum(data.num1) || verifyNum(data.num1)) {
+    console.log("Input provided is not a number")
+  }
+  else if (data.num2 === 0) {
+    console.log("Cannot divide by zero")
+  }
+  else if (sign === '+') {
+    printNum(data, data.num1 + data.num2)
+  } 
+  else if (sign === '-') {
+    printNum(data, data.num1 - data.num2)
+  } 
+  else if (sign === '*') {
+    printNum(data, data.num1 * data.num2)
+  } 
+  else if (sign === '/') {
+    printNum(data, data.num1 / data.num2)
+  } 
+  else if (sign === '%') {
+    printNum(data, data.num1 % data.num2)
+  } 
+  else if (sign === '**') {
+    printNum(data, data.num1 ** data.num2)
+  } 
+  else {
+    console.log("Operator not recognized")
+  }
+} 
+
+/// BASE CASE EXAMPLES ///
+
+console.log("*** ADDITION ***");
```

Comment:
You could consider adding this string (`**** ADDITION ****`, etc.) to each of your input example objects and then using your `printNum` function to console.log this string along with the mathematical expression